cosmo_tester/framework/util.py

```diff
@@ -741,3 +741,47 @@ def wait_for_execution(manager, execution, logger):
         )
     logger.info('Execution complete')
     return execution
+
+
+def _get_thing_by_name(_thing_name, _things):
+    if isinstance(_things, dict):
+        _things = _things.items()
+    for _thing in _things:
+        if _thing['name'] == _thing_name:
+            return _thing
+    raise Exception(
+        'No thing named {0} in {1}'.format(_thing_name, _things))
+
+
+def download_asset(repository_path,
+                   release_name,
+                   asset_name,
+                   save_location,
+                   git_token=None):
+
+    session = requests.Session()
+    if git_token:
+        session.headers['Authorization'] = 'token %s' % git_token
+    else:
+        username = os.environ['GITHUB_USERNAME']
+        password = os.environ['GITHUB_PASSWORD']
+        session.auth = (username, password)
+    releases = session.get(
+        'https://api.github.com/repos/{0}/releases'.format(
+            repository_path))
+    if not releases.ok:
+        raise Exception(
+            'Failed to authenticate to {0}, reason: {1}'.format(
+                releases.url, releases.reason))
+
+    release = _get_thing_by_name(release_name, releases.json())
+    asset = _get_thing_by_name(asset_name, release['assets'])
+    session.headers['Accept'] = 'application/octet-stream'
+
+    with session.get(asset['url'], stream=True) as response:
+        if not response.ok:
+            raise Exception(
+                'Failed to download {0}'.format(asset['url']))
+        with open(save_location, 'wb') as f:
+            for chunk in response.iter_content(1024):
```

Comment:
Is there a network limit?

Reply:
no

Reply:
So we can use regular requests.get, no?

Reply:
What chunk size do you prefer?

Reply:
We're downloading a file which is potentially GB in size.